src/DataGrid.tsx

```diff
@@ -729,7 +785,7 @@ function DataGrid<R, K extends keyof R, SR>({
         onSelectedRowsChange={onSelectedRowsChange}
         sortColumn={sortColumn}
         sortDirection={sortDirection}
-        onSort={onSort}
+        onSort={handleSort}
```

Comment:
Since `onSort` is not optional anymore for `HeaderRow` and its children, we should update the types.